camunda-bpm-camel-common/src/main/java/org/camunda/bpm/camel/component/CamundaBpmComponent.java

```diff
@@ -61,6 +68,7 @@ public ProcessEngine getProcessEngine() {
         return this.processEngine;
     }
 
+    @Autowired
```

Comment:
This looks strange to me - as I think the camel-common module should be independent of Spring (as it will be used in CDI and Spring contexts). So is this really necessary? Why did it work without in prior versions?

Reply:
I am not really sure how it worked previously. It might be related to how Spring instantiates the object.

Spring creates the CamundaBpmComponent (configured in META-INF.services.org.apache.camel.component) and chooses the default constructor. Without ProcessEngine as argument.

There are more Spring annotations in camel-common (abstract class CamelServiceCommonImpl).

I can remove the Autowired annotation and do one of the following:
- Remove the default constructor
- Create a Spring specific class which extends CamundaBpmComponent with one constructor which takes the ProcessEngine as an argument.

Reply:
Hi @berndruecker , I forgot to tag you in my reply.

Reply:
If there are already other Spring annotations available it should not harm. I will merge it (now with the CLA signed - thanks!)

Reply:
Thanks @berndruecker !